src/cmd/update.rs

```diff
@@ -0,0 +1,68 @@
+use std::fs;
+use std::fs::File;
+use std::fs::Permissions;
+use std::os::unix::prelude::*;
+
+use anyhow::Result;
+use serde::Deserialize;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const RELEASES_LIST: &str = "https://api.github.com/repos/sbstp/kubie/contents/releases/linux?ref=master";
+const RELEASE_BASE_URL: &str = "https://github.com/sbstp/kubie/raw/master/releases/linux/amd64";
+const FILENAME: &str = "kubie";
+
+#[derive(Debug, Deserialize)]
+struct TreeUrl {
+    git_url: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Tree {
+    tree: Vec<KubieVersion>,
+}
+
+#[derive(Debug, Deserialize)]
+struct KubieVersion {
+    path: String,
+}
+
+pub fn update() -> Result<()> {
+    let latest_version = get_latest_version()?;
+    if latest_version.eq(&format!("v{}", VERSION)) {
+        println!("Kubie is up-to-date : v{}", VERSION);
+    } else {
+        println!(
+            "A new version of Kubie is available ({}), the new version will be automatically installed...",
+            latest_version
+        );
+        let resp = attohttpc::get(format!("{}/{}/{}", RELEASE_BASE_URL, latest_version, FILENAME)).send()?;
+        if resp.is_success() {
+            let file = File::create(FILENAME)?;
+            let _n = resp.write_to(file)?;
+            let old_file = which::which(FILENAME).unwrap();
+            let res = replace_file(old_file.to_string_lossy().to_string(), FILENAME.to_string());
+            if res.is_ok() {
```

Comment:
Instead of using `is_ok()` and `err()` you should use pattern matching.


```rust
match res {
  Ok(_) => {
    println!("kubie has been updated");
  }
  Err(err) => {
    println!("Updated failed... {}", err);
  }
}
```

Reply:
You could even use anyhow's context here as well.

````rust
replace_file(...).context("The update failed, maybe consider using sudo")?;
```

Reply:
Done! I used pattern matching.